compiler/optimizer/LocalOpts.cpp

```diff
@@ -7462,18 +7461,12 @@ int32_t TR_InvariantArgumentPreexistence::perform()
                   }
                else
                   {
-                  clazz = fe()->getClassFromSignature(sig, len, feMethod);
+                  clazz = clazz ? clazz : fe()->getClassFromSignature(sig, len, feMethod);
```

Comment:
Should you be taking the more specific of clazz or getClassFromSignature ? i.e. maybe both are non-null but one provides "better" type info than the other (?)

Reply:
Inliner will try to refine argument types and put the result in argInfo. The type from argInfo should be the same as or more specific than the type from method symbol.

Reply:
should we have an assert for that then maybe just to verify it is true and warn ourselves is something breaks that invariant unintentionally later on?

Reply:
Assertion is added